test/cppsuite/tests/search_near_02.cxx

```diff
@@ -0,0 +1,430 @@
+                validate_successful_calls(
+                  cursor_default, cursor_prefix, prefix, exact_default, exact_prefix);
+            /* The prefix search near call failed. */
+            else
+                validate_unsuccessful_prefix_call(cursor_default, prefix, exact_default);
+        }
+    }
+
+    void
+    validate_successful_calls(scoped_cursor &cursor_default, scoped_cursor &cursor_prefix,
+      const std::string &prefix, int exact_default, int exact_prefix)
+    {
+        const char *k;
+        std::string k_str;
+        int ret;
+
+        /*
+         * The prefix search near call cannot retrieve a key with a smaller value than the prefix we
+         * searched.
+         */
+        testutil_assert(exact_prefix == 0 || exact_prefix == 1);
+
+        /* Retrieve the keys each cursor is pointing at. */
+        const char *key_default;
+        testutil_check(cursor_default->get_key(cursor_default.get(), &key_default));
+        std::string key_default_str = key_default;
+
+        const char *key_prefix;
+        testutil_check(cursor_prefix->get_key(cursor_prefix.get(), &key_prefix));
+        std::string key_prefix_str = key_prefix;
+
+        /* The key at the prefix cursor should contain the prefix. */
+        testutil_assert(key_prefix_str.substr(0, prefix.size()) == prefix);
+
+        logger::log_msg(LOG_TRACE,
+          "search_near (normal) exact " + std::to_string(exact_default) + " key " + key_default);
+        logger::log_msg(LOG_TRACE,
+          "search_near (prefix) exact " + std::to_string(exact_prefix) + " key " + key_prefix);
+
+        /* Example: */
+        /* keys: a, bb, bba */
+        /* Default search_near(bb) returns a, exact = -1 */
+        /* Prefix search_near(bb) returns bb, exact = 0 */
+        /* Prefix search_near(bb) returns bba, exact = 1 */
+        if (exact_default == -1) {
+            /* The key at the default cursor should not contain the prefix. */
+            testutil_assert((key_default_str.substr(0, prefix.size()) != prefix));
+
+            /*
+             * At least all the keys between the default cursor and the prefix cursor should contain
+             * the prefix.
+             */
+            while (ret = cursor_default->next(cursor_default.get()) == 0) {
+                testutil_check(cursor_default->get_key(cursor_default.get(), &k));
+                k_str = k;
+                testutil_assert((k_str.substr(0, prefix.size()) == prefix));
+                if (k_str == key_prefix_str)
+                    break;
+            }
+
+            /* We should be at the key pointed by the prefix cursor. */
+            testutil_assert(ret != WT_NOTFOUND && k_str == key_prefix_str);
+        }
+        /* Example: */
+        /* keys: a, bb, bba */
+        /* Default search_near(bb) returns bb, exact = 0 */
+        /* Prefix search_near(bb) returns bb, exact = 0 */
+        /* Prefix search_near(bb) returns bba, exact = 1 */
```

Comment:
Don't need to worry about it returning back "bba" either.

Reply:
Yes I agree, if the default search near returns `bb`, it means `bb` is visible and should be returned by the prefix search near.

Reply:
Fixed in a5602ced362d3cf2d7bf95b2e4b8469689f4265a